vignettes/variable-order.Rmd

```diff
@@ -16,99 +16,175 @@ load("vignettes.RData")
 
-## Viewing the order
+## File-system like
 
-Picking up with the dataset we used in the [array variables vignette](array-variables.html), we can fetch the `ordering` property of the dataset to view the current organization:
+This variable hierarchy can be thought of like a file system on your computer, with files (variables) organized into directories (folders). As such, the main functions you use to manage it are reminiscent of a file system.
 
-```{r, eval=FALSE}
-ordering(ds)
-```
-```{r, echo=FALSE}
-cat(step0.print, sep="\n")
+* `cd()`, changes directories, i.e. selects a folder
+* `mkdir()` makes a directory, i.e. creates a folder
+* `mv()` moves variables and folders to a different folder
+* `rmdir()` removes a directory, i.e. deletes a folder
+
+Like a file system, you can express the "path" to a folder as a string separated by a "/" delimiter, like this:
+
+```r
+mkdir(ds, "Brands/Cars and Trucks/Domestic")
 ```
 
-It's flat. If you're importing data from a `data.frame` or a file, like an SPSS file, this is where you'll begin.
+If your folder names should legitimately have a "/" in them, you can set a different character to be the path separator. See `?mkdir` or any of the other functions' help files for details. Paths can be expressed relative to the current object---a folder, or in this case, the dataset, which translates to its top-level `"/"` root folder in path specification---and the file system's special path segments `".."` (go up a level) and `"."` (this level) are also supported. We'll use those in examples below.
 
-## Creating groups
-Since I know how this dataset is organized, I'm going quickly toss the variables into one of a small number of groups, instantiated with `VariableGroup`, and collect them in a container object called `VariableOrder`.
+You can also specify paths as a vector of path segments, like
 
-```{r, eval=FALSE}
-ordering(ds) <- VariableOrder(
-        VariableGroup("Demos", ds[c(1, 21:37)]),
-        VariableGroup("Tracking questions", ds[c(2,3, 11:20)]),
-        VariableGroup("This week", ds[4:11])
-    )
+```r
+mkdir(ds, c("Brands", "Cars and Trucks", "Domestic"))
 ```
 
-Now, our variable tree has some structure:
+which is equivalent to the previous example. One or the other way may be more convenient, depending on what you're trying to accomplish.
 
-```{r, eval=FALSE}
-ordering(ds)
+These four functions all take a dataset or a folder as the first argument, and they return the same object passed to it, except for `cd`, which returns the selected folder. As such, they are designed to work with `magrittr`-style piping (`%>%`) for convenience in chaining together steps, though they don't require that you do.
+
+## Viewing the folders
+
+To get started, let's pick up the dataset we used in the [array variables vignette](array-variables.html) and view its starting layout. We can do that by selecting the root folder ("/") and printing it
+
+```r
+library(magrittr)
+ds %>%
+    cd("/") %>%
```

Comment:
It might be worth including a helper for `cd("/")`, I didn't find it super natural to use `cd` when I wanted to take a look at the variable folders for a dataset I think because my mind only went there when I wanted to change the directory. Another thought is we could modify the print method for datasets to include folder information, and so avoid the method collision between datasets and the top level VariableFolder.

Reply:
Yeah, I noticed this too. Changing the print method for CrunchDataset makes some sense, I think.